src/SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -216,40 +216,37 @@ private void PostProcessProjectStatus(IEnumerable<ProjectData> projects)
             }
         }
 
-        private static ProjectData GetClosestProjectOrDefault(string filePath, IEnumerable<ProjectData> projects)
+        internal /* for testing */ static ProjectData GetSingleClosestProjectOrDefault(FileInfo fileInfo,
```

Comment:
Please have a good review of this method (I added some specific tests) because I think the old behavior was somehow broken

Reply:
I remember seeing a SO question about an issue that looked like it might be due to the closest project being wrong (can't remember which one unfortunately - someone else answered it).

Out of interest, did you try running your new tests against the old code? It's not conclusive, but if the old tests failed and the tests now pass then we'd know for sure that there was a bug that has been fixed.

Reply:
I actually fixed the code because my first test was wrong (I mean test was not returning what I'd expect).

Reply:
FYI the old code is doing case-sensitive comparisons using _StartsWith_.

Reply:
Yes I saw! This was not returning the right project when nesting projects under each other (which should not happen in most cases).